### PR TITLE
feat(sass): Enable by default

### PR DESCRIPTION
**Pull request by Geobert**
Trying by keeping VS2017 and bump Rust stable version

### CODE REVIEW

**Comment by Geobert** [comment]
Ok, I figured out for the compilation, we were missing PLATFORM flag and --all, don't really know why.

Test wise, cli test are failing if we do `static ref _BIN: PathBuf = CWD.join("target/debug/cobalt");` but works if we do `static ref _BIN: PathBuf = CWD.join("target\\debug\\cobalt");` which does not make any sense as Rust is supposed to abstract this for us. 

I tried to remove lazy_static to see if it's tampering somehow, but we need it.

**Comment by Geobert** [comment]
And sometime, we have this error: https://ci.appveyor.com/project/johannhof/cobalt-rs/build/1.0.775

File: disable.travis.yml [hunk 1]
```diff
@@ -27,22 +27,22 @@ matrix:
     - cmake --version
     - cargo coverage || true
   - env: RUSTFMT
-    rust: 1.24.0  # `stable`: Locking down for consistent behavior
+    rust: 1.25.0  # `stable`: Locking down for consistent behavior
```
Comment (epage) [inline disable.travis.yml]: FYI If we move forward with this, I'd expect an explanation in the commit message and PR for why we need to bump our minimum version

File: appveyor.yml [hunk 2]
```diff
@@ -30,10 +33,10 @@ install:
 
 test_script:
   # we don't run the "test phase" when doing deploys
-  - if [%APPVEYOR_REPO_TAG%]==[false] ( cargo +%CHANNEL%-%TARGET% build )
-  - if [%APPVEYOR_REPO_TAG%]==[false] ( cargo +%CHANNEL%-%TARGET% test )
+  - if [%APPVEYOR_REPO_TAG%]==[false] ( cargo +%CHANNEL%-%TARGET% build --all )
```
Comment (epage) [inline appveyor.yml]: So there were a lot of commits with a lot of failures.  Do you happen to remember what failure you were resolving by adding this?

File: appveyor.yml [hunk 3]
```diff
@@ -13,14 +13,17 @@ environment:
     - TARGET: x86_64-pc-windows-msvc
       CHANNEL: stable
       TOOLCHAIN: msvc
+      PLATFORM: x86_64
 
     # Beta channel
     - TARGET: x86_64-pc-windows-msvc
       CHANNEL: beta
       TOOLCHAIN: msvc
+      PLATFORM: x86_64
```
Comment (epage) [inline appveyor.yml]: Yeah, in #398 I forgot to copy these over for some reason.

**Comment by epage** [comment]
> Test wise, cli test are failing if we do static ref _BIN: PathBuf = CWD.join("target/debug/cobalt"); but works if we do static ref _BIN: PathBuf = CWD.join("target\\debug\\cobalt"); which does not make any sense as Rust is supposed to abstract this for us.

That seems very suspect.  Is this locally or on appveyor?  Do you have a appveyor log to point to?

> And sometime, we have this error: https://ci.appveyor.com/project/johannhof/cobalt-rs/build/1.0.775
I got it twice in my trials but the second time it made no sense at all since the change was only on cli.rs

Maybe there is something here to help
https://github.com/rust-lang/rust/issues/33145

File: disable.travis.yml (hunk 1 above)
Comment (Geobert) [reply disable.travis.yml]: Oh, I'm the kind of guy that always try to use the latest stable to benefit from code generation improvement and performance. As I'm on 1.25, I noticed that rustfmt behave differently. If we don't bump, I'll install 1.24 for cobalt locally :)

File: appveyor.yml (hunk 2 above)
Comment (Geobert) [reply appveyor.yml]: I'll track back the errors, I forgot to keep a link -_-

File: disable.travis.yml (hunk 1 above)
Comment (epage) [reply disable.travis.yml]: Oh, right, this is just for locking down rustfmt for consistent output.  Similar for warnings.,  Ideally these would be upgraded in an independent PR.

**Comment by Geobert** [comment]
ok using VS2015 we have this random errors, that' s why I've updated to VS2017 when activating syntec on all platform.

The issue is that sass-sys does not build with VS2017. I reported the issue: https://github.com/compass-rs/sass-rs/issues/22#issuecomment-387876161

**Comment by epage** [comment]
Thanks for doing that investigation!

**Comment by Geobert** [comment]
Another solution is to see if my PR on libsass is accepted: https://github.com/sass/libsass/pull/2648

**Comment by epage** [comment]
> and it's true that gutenberg has many submodule compiled independantly :-/ Is that something we were planning to do?

I'm trying to avoid splitting up for the sake of compile times.

We already want to split jekyll out.  I think I want to split cobalt_model out but was waiting on further refactorings to better define what is or isn't in cobalt_model.

**Comment by Geobert** [comment]
ok, let's hope my PR will make it

**Comment by Geobert** [comment]
Another lead (long term) is this project: https://github.com/kaj/rsass

**Comment by Geobert** [comment]
I'm trying different leads: https://github.com/compass-rs/sass-rs/pull/25

this can also make sass-rs build on VS2107

**Comment by Geobert** [comment]
I'm waiting for the merge of https://github.com/compass-rs/sass-rs/pull/26 to update our dependency to sass-rs and we should be done with this at last!

**Comment by epage** [comment]
> I'm waiting for the merge of compass-rs/sass-rs#26 to update our dependency to sass-rs and we should be done with this at last!

Awesome job owning this to completion!

**Comment by Geobert** [comment]
My PR has been merged, waiting for a release with it now.
Can be delayed because of https://github.com/rust-lang/rust/issues/50778 though :-/

**Comment by Geobert** [comment]
Testing how Rust 1.27 behaves on appveyor, I didn't see the error on my computer for a while

**Comment by Geobert** [comment]
It works! 🎉 🎈

**Comment by epage** [comment]
My one concern is if this will stay working :(

**Comment by epage** [comment]
Now that we think we got this, could you clean up the history?

**Comment by Geobert** [comment]
I'll do a loop test tomorrow on my computer first ^^

**Comment by Geobert** [comment]
If I squash the merge, I have a conflict with `master` :-/ I think we can merge this, 1.27 may have solved the linker issue.

**Comment by epage** [comment]
I'd rather not have merges in PRs.

**Comment by Geobert** [comment]
How do I solve this?

**Comment by epage** [comment]
Have you tried to rebase your branch against master locally?

**Comment by Geobert** [comment]
yay! Thanks! :)

**Comment by Geobert** [comment]
I'm not use to rebase yet, need to read a bit more on this…

**Comment by epage** [comment]
Thanks!

**Pull request merged**